src/serializer/ResidualFunctions.js

```diff
@@ -570,6 +562,14 @@ export class ResidualFunctions {
       }
     }
 
+    for (let [additionalFunction, body] of additionalFunctionPreludes.entries()) {
```

Comment:
Since ordering matters now, maybe add a comment documenting this ordering so that future changes will not try to revert it back.

Reply:
There's a test case now that would expose the issue if someone changes it...